File: LeetCode-December-Challenge/SortCharactersByFrequency.js
var frequencySort = function (s) {
    let freq = {};
    for (let ch of s)
        freq[ch] ? freq[ch]++ : freq[ch] = 1;
    let arr = [];
    for (let x in freq)
        arr.push([freq[x], x])
    arr.sort((a, b) => b[0] - a[0]);
    let res = "";
    for (let x of arr)
        for (let i = 0; i < x[0]; ++i)
            res += x[1];
    return res;
};
console.log(frequencySort("tree"));

File: Year-2022/LeetCode-December-Challenge/SortCharactersByFrequency.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    string frequencySort(string s) {
        unordered_map<char, int> mp;
        for(auto ch: s) mp[ch]++;
        vector<pair<char, int>> vec;
        for(auto ch: mp) vec.push_back({ch.first, ch.second});
        sort(vec.begin(), vec.end(), [](auto &a, auto &b){return a.second > b.second;});
        string res = "";
        for(auto ch: vec)
            for(int i = 0; i < ch.second; ++i)
                res += ch.first;
        return res;
    }
};
int main(){
    string s = "tree";
    Solution *S = new Solution();
    cout<<S->frequencySort(s)<<endl;
    return 0;
}